### PR TITLE
[JENKINS-60409, JENKINS-60199] - Jenkins 2.204.x: Revert Winstone from 5.8 to 5.4.1 with backports

### PR DESCRIPTION
**Pull request by oleg-nenashev**
This change picks up https://github.com/jenkinsci/winstone/pull/96 with the backports for the LTS. This is an alternative to #4542 for the LTS baseline

See [JENKINS-XXXXX](https://issues.jenkins-ci.org/browse/JENKINS-XXXXX).

### Proposed changelog entries

* JENKINS-60409, JENKINS-60199 - Downgrade Winstone from 5.8 to 5.4.1 to fix regressions caused by newer Jetty versions. High default maximum form size limit and reverse proxy redirection are restored (regressions in 2.204.3)
  * Winstone 5.4.1 changelog: https://github.com/jenkinsci/winstone/releases/tag/winstone-5.4.1

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] JIRA issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a JIRA issue should exist and be labeled as `lts-candidate`

### CODE REVIEW

**Review by jglick — DISMISSED** [review]

**Review by markjacksonfishing — APPROVED** [review]

**Review by MarkEWaite — APPROVED** [review]
I assume this will switch from the incrementals release to the 5.4.1 release once the 5.4.1 release is available.

**Review by res0nance — DISMISSED** [review]

**Review by MarkEWaite — DISMISSED** [review]

**Review by batmat — DISMISSED** [review]

**Comment by everops-alex** [comment]
@oleg-nenashev you're a legend, my friend. Thanks for all of your work! 🙇

**Review by daniel-beck — CHANGES_REQUESTED** [review]
The form size change was introduced in Jetty 9.4.20, this uses 9.4.22, and the PR https://github.com/jenkinsci/winstone/pull/96 does not include Jesse's winstone-side fix.

So this does not appear to address JENKINS-60409.

**Review by daniel-beck — CHANGES_REQUESTED** [review]
5.4.2 appears to still be affected by JENKINS-60199, introduced in Jetty 9.4.20 and fixed in 9.4.27.

File: war/pom.xml [hunk 1]
```diff
@@ -101,7 +101,7 @@ THE SOFTWARE.
       -->
       <groupId>org.jenkins-ci</groupId>
       <artifactId>winstone</artifactId>
-      <version>5.8</version>
+      <version>5.4.2</version>
```
Comment (batmat) [inline war/pom.xml]:
```suggestion
      <version>5.4.3</version>
```

**Comment by batmat** [comment]
Just bumped to 5.4.3 just released by @olamy

**Comment by batmat** [comment]
Will test the incrementals once out.

**Comment by olivergondza** [comment]
I lean towards getting back to winstone 5.3. This would regress [JENKINS-60857](https://issues.jenkins-ci.org/browse/JENKINS-60857) that was fixed in .3, but it is less sever than eiter of [JENKINS-60857](https://issues.jenkins-ci.org/browse/JENKINS-60857), [JENKINS-60409](https://issues.jenkins-ci.org/browse/JENKINS-60409) or [JENKINS-60199](https://issues.jenkins-ci.org/browse/JENKINS-60199).

Am I missing something?

**Comment by daniel-beck** [comment]
@olivergondza Poked you on IRC, let's go real time.

**Comment by olivergondza** [comment]
@oleg-nenashev, @jglick, do you object against my earlier proposal?

**Comment by olivergondza** [comment]
Oleg, as agreed, I am closing this one in favor of [revert to 5.3](4b6da0c3e4dfbb402d86f7a5124ca6e323b2c959) I have just backported. Thanks you everyone for the hard work trying to nail this one (several ones, in fact) down, but I felt strongly that we should stand on the conservative side.

**Comment by oleg-nenashev** [comment]
Sorry, I missed this conversation while discussing the same topic in IRC and other chats. I agree with the selected strategy. Should we had more time, we could have created 5.3.1 with JENKINS-57888 backporting, but that regression is much less important than the mess the backporting of it caused. We can always cut 2.204.6, but maybe it's an overkill since 2.222.1 arrives in 2 weeks.

I have also started  a public retrospective document for 2.204.3...2.204.5 here: https://docs.google.com/document/d/1NzR1XtkCfk6MDSD1jRq5H-iPSjqKvP-pfUO0b9lVZ9k/edit?usp=sharing

**Pull request closed without merging**